app/code/Magento/Payment/Model/Method/Adapter.php

```diff
@@ -507,12 +507,13 @@ private function executeCommand($commandCode, array $arguments = [])
             return null;
         }
 
+        $payment = $arguments['payment'];
```

Comment:
@evgk, thank you for catching and fixing  this issue.

According to code below it's possible that `$arguments['payment']` would be not defined so notice may be raised. Please modify code so it will handle such cases correctly